mne/preprocessing/bads.py

```diff
@@ -38,3 +38,176 @@ def find_outliers(X, threshold=3.0, max_iter=2):
 
     bad_idx = np.where(my_mask)[0]
     return bad_idx
+
+
+def corrmap(icas, template, 
+            threshold="auto", 
+            name="marked_ics", 
+            verbose_return=False, 
+            plot=True,
+            inplace=False):
+    
+    """Corrmap (Viola et al. 2009 Clin Neurophysiol) identifies the best group
+    match to a supplied template. Typically, feed it a list of fitted ICAs and
+    a template IC, for example, the blink for the first subject, to identify
+    specific ICs across subjects.
+    
+    The specific procedure consists of two iterations. In a first step, the maps 
+    best correlating with the template are identified. In the step, the analysis
+    is repeated with the mean of the maps identified in the first stage.
+    
+    Outputs a list of fitted ICAs with the indices of the marked ICs in a 
+    specified field.
+    
+    The original Corrmap website: www.debener.de/corrmap/corrmapplugin1.html
+    
+    Parameters
+    ----------
+    icas : list
+        A list of fitted ICAs.
+    template : (int, int) tuple
+        Index of the list member from which the template should be obtained, and
+        the index of the IC.
+    threshold : "auto" | list of floats > 0 < 1 | float > 0 < 1 | float > 1
+        Correlation threshold for identifying ICs
+        If "auto": search for the best map by trying all correlations between 
+        0.6 and 0.95. In the original proposal, lower values are considered, but
+        this is not yet implemented.
+        If list of floats: search for the best map in the specified range of 
+        correlation strengths.
+        If float > 0: select ICs correlating better than this.
+        If float > 1: use find_outliers to identify ICs within subjects (not in 
+        original Corrmap)
+        Defaults to "auto".
+    name : string
+        Name of the field under which found ICs should be specified.
+        Defaults to "marked_ics".
+    verbose_return : bool
+        Should the output be restricted to the list of ICs (False), or also the 
+        correlation of the constructed template with the selected maps and the 
+        correlations?
+        Defaults to False.
+    plot : bool
+        Should constructed template and selected maps be plotted?
+    ----------
+    Returns : list 
```

Comment:
.. which I do not find necessarily more inuitive or cleaner. Anyways such an attribute it will create a certain redundeancy with .exclude ... but tha'ts probably acceptable as long as the purpose is clean. Maybe .classified or something along that lines? @agramfort any gut feeling?